src/super_gradients/training/models/segmentation_models/ddrnet_backbones.py

```diff
@@ -0,0 +1,73 @@
+from typing import Tuple
+
+import torch
+
+from super_gradients.training.models.segmentation_models.ddrnet import DDRNet39
+from super_gradients.training.utils import HpmStruct
+
+__all__ = ["DDRNet39Backbone"]
+
+
+class DDRNet39Backbone(DDRNet39):
```

Comment:
Why not inherit from super_gradients.training.models.segmentation_models.ddrnet.BasicDDRBackBone ?
Im feeling a bit unease with the amount of duplicate backbone classes we have.

Reply:
I actually didn't notice it exists 😅 because of quite massive inheritance chain there.
A BasicDDRBackBone has a bunch of it's own problems. We instantiate and put the whole backbone into arch_params o_O in order store it later in base class. 

```
class DDRNet23Slim(DDRNetCustom):
    def __init__(self, arch_params: HpmStruct):
        _arch_params = HpmStruct(**DEFAULT_DDRNET_23_SLIM_PARAMS)
        _arch_params.override(**arch_params.to_dict())
        # BUILD THE BACKBONE AND INSERT TO THE _arch_params
        backbone_layers, _arch_params.additional_layers = _arch_params.layers[:4], _arch_params.layers[4:]
        _arch_params.backbone = BasicDDRBackBone(  <---- HERE
            block=_arch_params.block,
            width=_arch_params.planes,
            layers=backbone_layers,
            input_channels=_arch_params.input_channels,
            layer3_repeats=_arch_params.layer3_repeats,
        )
        super().__init__(_arch_params)
```

Reply:
Well, I don't seem to see the problem yet - we are talking about inheriting the BasicDDRBackBone and actually its your implementation that inherits from DDRNet itself. What am I missing ?

Reply:
Unfortunately it is not that easy to go with BasicDDRBackBone and here's why
1) BasicDDRBackBone  does not have forward() method implemented at all. So we have to add one. Not a big deal as we still have our own forward method anyway
2) BasicDDRBackBone  does not have `layer5` and `spp` layer (They are initialized in derived class). That is more serious as we will have to instantiate them. During construction of layer5 and spp layer we actually do forward() to get number of output channels
```
        self._backbone = backbone
        self._backbone.validate_backbone_attributes()
        out_chan_backbone = self._backbone.get_backbone_output_number_of_channels() <---- HERE
```

To me the right way to get this done - refactor DDRNet entirely to extract more clear backbone class and build DDRnetSegmentation / DDRNetClassification models on top of it. Without this all attempts to use ddrnet backbone would look ugly

Reply:
mmm to that I agree.
This is a recurring issue, so perhaps we should have a convo about re-designing the base modules to prevent further discussions like this one (for example, we can introduce new base classes and have them inherit alongside with CustomizableDetector from some base class, something like CustomizableModel).
We should consider taking some time for refactoring before we integrate more models into SG. Lets merge this though, as it is clearly out of the scope here...